lib/ansible/module_utils/urls.py

```diff
@@ -866,7 +866,7 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
              force=None, last_mod_time=None, timeout=None, validate_certs=None,
              url_username=None, url_password=None, http_agent=None,
              force_basic_auth=None, follow_redirects=None,
-             client_cert=None, client_key=None, cookies=None):
+             client_cert=None, client_key=None, cookies=None, auth_realm=None):
```

Comment:
We should add `auth_realm` to `Request.__init__` as well.  This will also require a new `self._fallback` call a little further down in this function (`Request.open`).

Please also document this new arg in the docstring for `Request.open` also.

Tests will need to be updated to account for the additional call to `_fallback`.

Reply:
`auth_realm` should also be added to `fetch_url`.